### PR TITLE
[testlib] Make GPU burn test a library test

### PR DESCRIPTION
**Pull request by vkarak**
More particularly, this PR offers the following:

- Converts the GPU burn test into a library test.
- Bases the the CSCS test on this new library test.
- Introduces new command line arguments to the GPU burn test itself, so as to control better its behavior.
- Introduces a new source file organisation for the GPU microbenchmarks (currently, only the GPU burn test follows the new structure, but the rest of the tests will have to follow it as soon as they are updated).
- The CSCS config is updated with metadata on its partitions and environments.
- A new convenience method is added to the `SystemPartition` that allows to select the devices of a certain type.

### CODE REVIEW

File: config/cscs.py [hunk 1]
```diff
@@ -39,85 +39,81 @@
...
+                            'num_devices': 2
+                        }
+                    ]
                 },
                 {
                     'name': 'amdvega',
                     'scheduler': 'slurm',
                     'access': ['-pamdvega'],
-                    'environs': ['builtin', 'PrgEnv-gnu'],
+                    'environs': ['gnu', 'rcom'],
```
Comment (rsarm) [inline config/cscs.py:76]:
```suggestion
                    'environs': ['gnu', 'rocm'],
```

**Review by rsarm — APPROVED** [review]

File: hpctestlib/microbenchmarks/gpu/gpu_burn.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    def valid_build(self):
+        return True
+
+
+@rfm.simple_test
+class gpu_burn_check(rfm.RunOnlyRegressionTest):
+    '''GPU burn benchmark.
+
+    This benchmark runs continuously GEMM, either single or double precision,
+    on a selected set of GPUs on the node that it runs.
```
Comment (teojgo) [inline hpctestlib/microbenchmarks/gpu/gpu_burn.py:94]:
```suggestion
    on a selected set of GPUs of the node on which it executes.
```

File: hpctestlib/microbenchmarks/gpu/gpu_burn.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+    The floating point precision of the computations, the duration of the
+    benchmark as well as the list of GPU devices that the benchmark will run
+    on can be controlled through test variables.
+
+    This benchmark tries to build the benchmark code through the
+    :class:`gpu_burn_build` fixture.
+
+    This benchmark does sets the
```
Comment (teojgo) [inline hpctestlib/microbenchmarks/gpu/gpu_burn.py:103]:
```suggestion
    This benchmark sets the
```

File: hpctestlib/microbenchmarks/gpu/gpu_burn.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    :attr:`~reframe.core.pipeline.RegressionTest.num_gpus_per_node` test
+    attribute, if not already set, based on the number devices with ``type ==
```
Comment (teojgo) [inline hpctestlib/microbenchmarks/gpu/gpu_burn.py:105]:
```suggestion
    attribute, if not already set, based on the number of devices with ``type ==
```

**Review by teojgo — CHANGES_REQUESTED** [review]

File: hpctestlib/microbenchmarks/gpu/gpu_burn.py (hunk 2 above)
Comment (vkarak) [reply hpctestlib/microbenchmarks/gpu/gpu_burn.py:94]: I rephrased it a bit differently.

**Review by teojgo — APPROVED** [review]

**Pull request merged**